### ISSUE TITLE
[BUG] Push notifications keep arriving after logout

### ISSUE THREAD

**Issue by fnkr** [RocketChat/Rocket.Chat.Android#336]
Rocket.Chat.Android version: 1.0.14

Seems like I'm not the only one: https://github.com/RocketChat/Rocket.Chat.Android/issues/77#issuecomment-305454719

**Comment by valerionew** [comment]
+1 
Happens to me too

**Comment by rafaelks** [comment]
That's probably because we're not clearing the token into the server when user logouts. Can you confirm @leonardoaramaki, please?

**Comment by geekgonecrazy** [comment]
I can confirm :)

**Comment by rafaelks** [comment]
@rodrigok @SeanPackham Can you guys update the Real Time API with logout method, so we can call it on the native apps when user tries to logout? I assume that this call should clear the push token.

Current docs is blank: https://rocket.chat/docs/developer-guides/realtime-api/method-calls/logout/#logout

**Comment by geekgonecrazy** [comment]
```
["{\"msg\":\"method\",\"method\":\"logout\",\"params\":[],\"id\":\"16\"}"]
```
that's the raw query that goes over the websocket

So pretty much call method: `logout` with no params and it'll log you out

**Comment by geekgonecrazy** [comment]
That call I don't think will remove your push token though.

Looking at the methods we have available... it looks like raix:push strangely does not have a way to remove push tokens... https://github.com/raix/push/blob/master/lib/server/server.js

**Comment by luciofm** [comment]
We should check if the user is logged in on the Push Receiver on the App, we should also check if the push is for the current user, let's say the User logout from one username and logins with another, he should not receive (or at least show notifications) for the old username.

**Comment by geekgonecrazy** [comment]
Yes on the Android side that would for sure work, long term we should also try and get the ability to remove a token too.

**Comment by rafaelks** [comment]
Yes, this is something we need to handle for both iOS and Android

@rodrigok Do we have a way to clear the device token from the server? On iOS, we do save the device identifier (https://github.com/RocketChat/Rocket.Chat.iOS/blob/ff09823ed7ab0672519cacc14913cc75b2b203f8/Rocket.Chat/Managers/PushManager.swift#L24), so in theory, we could remove this from the server with a simply API call.

**Comment by geekgonecrazy** [comment]
I think we could definitely open a PR against raix:push to add a method to remove.  But currently there isn't a way.

https://github.com/raix/push/issues/298
https://github.com/raix/push/issues/61

According to this comment: https://github.com/raix/push/issues/61#issuecomment-92314165 you might be able to update the token and invalidate it, which would cause it to be remove.  So that might be an option.

**Comment by rafaelks** [comment]
@geekgonecrazy I think updating to an invalid token is OK. Maybe we set a standard value now, like `unknown` so in the future we can clear it?

**Comment by rafaelks** [comment]
Closing now, because it's already implemented in [2.0.0](https://github.com/RocketChat/Rocket.Chat.Android/milestone/6).
